src/error.rs

```diff
@@ -14,6 +14,7 @@ pub(crate) enum ErrorKind {
     Start,
     Join,
     BrokenPipe,
+    IncorrectArgument
```

Comment:
I think this is a just fine error but should be renamed `UnexpectedRequestType` to represent that we got an unexpected request type.